### PR TITLE
Update zombie soldier drops

### PR DESCRIPTION
**Pull request by ymber**
#### Summary
SUMMARY: Content "Update zombie soldier drops"

#### Purpose of change
Zombie soldiers were dropping all kinds of random things they shouldn't have and missing things that are standard issue.

#### Describe the solution
Modernize some itemgroups, update zombie soldier drops.

#### Testing
Spawning and killing 120 zombie soldiers gives the expected item distribution. Each individual zombie soldier has regular expected drops.

#### Additional context
I'd like to add `damage: [ 0, 1 ]` to some of these but I'm waiting on #26473 being fixed.

### CODE REVIEW

**Comment by anothersimulacrum** [comment]
Looks like you have changes from #36377 in here

File: data/json/itemgroups/military.json [hunk 1]
```diff
@@ -1,4 +1,89 @@
...
+    "entries": [ { "item": "m249", "prob": 100, "charges": [ 0, 200 ] } ]
+  },
+  {
+    "type": "item_group",
+    "id": "military_standard_pistols",
+    "subtype": "distribution",
+    "entries": [
+      { "item": "m17", "prob": 95, "charges": [ 0, 17 ] },
+      { "item": "glock_19", "prob": 3, "charges": [ 0, 15 ] },
+      { "item": "m1911_MEU", "prob": 2, "charges": [ 0, 7 ] }
```
Comment (LaVeyanFiend) [inline data/json/itemgroups/military.json]: Where's the M9? That's the direct predecessor to the M17, which aren't planned to be fully delivered until 2027(https://www.armytimes.com/news/your-army/2017/01/20/11-things-you-need-to-know-about-the-army-s-new-handgun/). 😕

**Comment by SirPendrak** [comment]
Modernising code is cool, adding missing helmets and pads is also good, but there are some problems: 
- caseless rifles will become much, much more rare. Is it intended? Because of that their ammo will be a rarity as well. No more random rifle with 500 8mm shells
- because laser sights are no longer craftable and soldiers drop them no more, they will be almost unobtainable. You will be more likely to find 20 rifle scopes before you get one uderbarrel lassight. 
- once you killed like 10 zsoldiers you probably cant get anything new important from them because of how similar gear they now carry. Unless you farm MREs or STANAGs for some reason.

**Comment by anothersimulacrum** [comment]
I don't think 1 or 3 are problems, 2 is not a problem with this (if it is a problem), it's a problem with where/how laser sights spawn.

**Comment by LaVeyanFiend** [comment]
Concern 2 could be remedied by making them drop their guns with attached sights on some of them using a setup like this one from #34638:

```
{
    "type": "item_group",
    "id": "guns_shotgun_rare_readied",
    "subtype": "distribution",
    "entries": [
      { "group": "guns_shotgun_rare", "prob": 22, "contents-group": "sights_shotgun_readied" },
      { "item": "winchester_1887", "prob": 10, "charges-min": 0, "charges-max": 6 },
      { "item": "winchester_1897", "prob": 7, "charges-min": 0, "charges-max": 6 }
    ]
  }
```
```
  {
    "type": "item_group",
    "id": "sights_shotgun_readied",
    "subtype": "distribution",
    "//": "For non-static shotguns (not in stores, closets ).",
    "entries": [ { "group": "sights_shotgun", "prob": 50 }, { "group": "EMPTY_GROUP", "prob": 50 } ]
  },
```
```
{
    "type": "item_group",
    "id": "sights_shotgun",
    "//": "Aftermarket sights that are for shotguns.",
    "items": [
      [ "improve_sights", 84 ],
      [ "red_dot_sight", 42 ],
      [ "holo_sight", 18 ],
      [ "laser_sight", 15 ],
      [ "offset_sights", 6 ]
    ]
  },
```

**Comment by ymber** [comment]
Laser sights are uncommon in military use. You should be much more likely to find one somewhere that sells them commercially. Experimental caseless rifles should be much rarer and definitely not in military service. These things drop food, clean water, ammunition, medical supplies, gas mask filters, grenades, and military IDs as well as being a source of kevlar so you've got plenty reason to keep hunting them throughout the game.

**Comment by tenmillimaster** [comment]
> Laser sights are uncommon in military use.

 *_visible_ laser sights. IR lasers will need a lot more doing, so in the meantime, their removal is the correct choice.

**Comment by SirPendrak** [comment]
Then maybe Bio-operators should get similar rework, and some of the fictional firearms could be moved to their drops. They are like 3x as dangerous and their drops are often underwhelming. And then give their guns like 90% chance to be modded and tacticool. 
Right now they can drop some fancy stuff, more often than old regular soldiers did, but maybe it is nice idea to make them distinctive by making them drop mostly modern-scifi equipement?

**Comment by thethunderhawk** [comment]
@SirPendrak brings up some interesting questions regarding Bio Operators. 

-Are they from a specific fictional unit of enhanced troops, or are they from normal IRL special operations units whose troops have been enhanced? If they're from a normal special operations unit under USSOCOM, with traditional logistics, that's going to limit the equipment they have access to. The most expensive and advanced field-tested weapons, yes. Experimental, possibly dangerous sci-fi stuff no.

- If they're from a specific unit, how closely related is that unit to the research programs which are developing the sci-fi gear? If these enhanced troops are using laser pistols and fusion rifles, that'd imply they can source their supplies directly from whatever agencies are in charge of working on those things. 

-Perhaps the "Bio Operator" mob represents enhanced troops from any of multiple units, some of which are more secretive (and have access to better technology) than others. That way you could justify the largest variety of gear. 

Depending on where the lore goes, the Bio Operators should drop different gear. But in general I agree that it'd be best for the game if they tended to drop the fancier weapons (Though IMO the most sci-fi of weapons should be found only in labs, but reasonable people could disagree about that.)

**Comment by Amoebka** [comment]
Am I missing something? The update seems to completely remove Rivtech weapons from bio-operators as well, does it not?

**Comment by thethunderhawk** [comment]
@amoebka looks to me like it basically aligns Bio-operator drops with regular soldier drops. Seems wrong to me, but I figure working out what Bio Operators are and what gear they should have ought to happen in a separate PR anyway.

**Comment by kevingranade** [comment]
Bio-operators are going to be most closely aligned with SOCOM type organizations, as in the gear might be a *shade* more cutting edge, but it doesn't make the jump from conventional to sci-fi, with the exception of the CBMs, which we're in the process of reevaluating.

**Comment by thethunderhawk** [comment]
In that case leaving it how it is should actually be pretty close to the goal for Bio-ops. One big difference @SirPendrak pointed was was that special operations troops are going to have nice sights on their rifles, they might be carrying especially nice ammo, and they’ll likely be adding ergonomic bits to their weapons. They could also have handcuffs, breaching tools, demolitions gear, surveillance equipment, rare types of grenade and shotgun rounds, incendiaries, sophisticated or miniaturized survival gear, (what else?)

**Pull request merged**